website/static/js/pages/home-page.js

```diff
@@ -6,13 +6,116 @@
 
 var $ = require('jquery');
 var m = require('mithril');
+var bootbox = require('bootbox');
+
 
 var QuickSearchProject = require('js/home-page/quickProjectSearchPlugin');
 var NewAndNoteworthy = require('js/home-page/newAndNoteworthyPlugin');
 var MeetingsAndConferences = require('js/home-page/meetingsAndConferencesPlugin');
 
 var columnSizeClass = '.col-md-10 col-md-offset-1 col-lg-8 col-lg-offset-2';
 
+
+var Raven = require('raven-js');
+var jstz = require('jstimezonedetect');
+var bootbox = require('bootbox');
+
+var $osf = require('js/osfHelpers');
+require('loaders.css/loaders.min.css');
+
+var confirmedEmailURL = window.contextVars.confirmedEmailURL;
+var removeConfirmedEmailURL = window.contextVars.removeConfirmedEmailURL;
+
+//choose which emails/users to add to an account and which to deny
+function confirm_emails(emails) {
+    if (emails.length > 0) {
+        var email = emails.splice(0,1)[0];
+        var title;
+        var requestMessage;
+        var confirmMessage;
+        var nopeMessage;
+        if (email.user_merge) {
+            title =  'Merge account';
+            requestMessage = 'Would you like to merge \<b>' + email.address + '\</b> into your account?  ' +
+                'This action is irreversable.';
+            confirmMessage = '\<b>' + email.address + '\</b> has been merged into your account.';
+            nopeMessage =  'You have chosen to not merge \<b>'+ email.address + '\</b>  into your account. ' +
+                'If you change your mind, visit the \<a href="/settings/account/">user settings page</a>.';
+        }
+        else {
+            title = 'Add email';
+            requestMessage = 'Would you like to add \<b>' + email.address + '\</b> to your account?';
+            confirmMessage = '\<b>' + email.address + '\</b> has been added into your account.';
+            nopeMessage = 'You have chosen not to add \<b>' + email.address + '\</b> to your account.' +
+             'If you change your mind, visit the \<a href="/settings/account/">user settings page</a>.';
+        }
+
+        var failSuccessMessage = 'There are a problem adding \<b>' + email.address +
+            '\</b>. Please contact <a href="mailto: support@osf.io">support@osf.io</a> if the problem persists.';
+
+        var failCancelMessage = 'There are a problem removing \<b>' + email.address +
+            '\</b>. Please contact <a href="mailto: support@osf.io">support@osf.io</a> if the problem persists.';
+
+        bootbox.dialog({
+            title: title,
+            message: requestMessage,
+            onEscape: function() {},
+            backdrop: true,
+            closeButton: true,
+            buttons: {
+                confirm: {
+                    label: 'Add email',
+                    className: 'btn-success',
+                    callback: function () {
+                        $osf.putJSON(
+                            confirmedEmailURL,
```

Comment:
Ah, I just saw that PUT is supported by a duplicate `Rule`. See https://github.com/CenterForOpenScience/osf.io/pull/5528/files#r61086751